yb-voyager/cmd/live_migration.go

```diff
@@ -135,14 +140,103 @@ func streamChanges(state *ImportDataState, tableNames []sqlname.NameTuple) error
 		}
 		log.Infof("got next segment to stream: %v", segment)
 
-		err = streamChangesFromSegment(segment, evChans, processingDoneChans, eventChannelsMetaInfo, statsReporter, state)
+		err = streamChangesFromSegment(segment, evChans, processingDoneChans, eventChannelsMetaInfo, statsReporter, state, tableToPartitioningStrategyMap)
 		if err != nil {
 			return fmt.Errorf("error streaming changes for segment %s: %v", segment.FilePath, err)
 		}
 	}
 	return nil
 }
 
+const (
+	PARTITION_BY_PK    = "pk"
+	PARTITION_BY_TABLE = "table"
+)
+
+func handleCdcPartitioningStrategy(tableNames []sqlname.NameTuple) (*utils.StructMap[sqlname.NameTuple, string], error) {
+	tableToPartitioningStrategyMap := utils.NewStructMap[sqlname.NameTuple, string]()
+
+	if importerRole != TARGET_DB_IMPORTER_ROLE {
+		// source or source-replica importer
+		msr, err := metaDB.GetMigrationStatusRecord()
+		if err != nil {
+			return nil, fmt.Errorf("error getting migration status record: %w", err)
+		}
+		switch msr.SourceDBConf.DBType {
+		case ORACLE:
+			//For oracle old behaviour same as partitioning by pk
+			for _, t := range tableNames {
+				tableToPartitioningStrategyMap.Put(t, PARTITION_BY_PK)
+			}
+		case POSTGRESQL:
+			//For pg partitioning by table since there is no huge difference in performance between the two strategies
+			//and Parititon by table is better from data correctness perspective
+			for _, t := range tableNames {
+				tableToPartitioningStrategyMap.Put(t, PARTITION_BY_TABLE)
+			}
+		}
+		return tableToPartitioningStrategyMap, nil
+	}
+
+	// target db importer
+	//fetch and check if metadb key IMPORT_DATA_STATUS_KEY is present and has TableToPartitioningStrategyMap
+	importDataStatus, err := metaDB.GetImportDataStatusRecord()
+	if err != nil {
+		return nil, fmt.Errorf("error getting cdc partitioning strategy: %w", err)
+	}
+	if importDataStatus != nil && importDataStatus.TableToPartitioningStrategyMap != nil {
```

Comment:
we don't expect importDataStatus record to be nil, right? We should fail in that case.